client/src/store/historyStore/historyItemsStore.js

```diff
@@ -47,9 +51,13 @@ const actions = {
     fetchHistoryItems: async ({ commit, dispatch }, { historyId, offset, filterText }) => {
         dispatch("startHistoryChangedItems", { historyId: historyId });
         const queryString = getQueryString(filterText);
+        const queryStats = "&view=count";
```

Comment:
This could have a significant impact on performance. Have you tested this on large histories ? Could this ever run on non-orm filters, where the ORM has to fetch every item from the database ?

Reply:
> Could this ever run on non-orm filters, where the ORM has to fetch every item from the database ?

It looks like we'd discard any function filter in `contents_count` level, which is good for performance. genome_build, data_type and annotation are potentially valid filters though that are not being applied, so that could lead to wrong counts here. Not a big deal since I think we don't expose these in the UI, but I really think we need to pull out the filter parsing and fail the request if these are included.

Reply:
I've tried just with 1000 datasets in the history, probably not large enough...

> Could this ever run on non-orm filters, where the ORM has to fetch every item from the database ?

To be honest, I was blindly trusting [content_counts](https://github.com/galaxyproject/galaxy/blob/7c442d63dfb91c748ee1ae3c0843fdc9a1543012/lib/galaxy/managers/history_contents.py#L120) to be the most optimised and tested way of counting all matching elements, since it was also used in https://github.com/galaxyproject/galaxy/pull/11436/files#diff-d7789a8b5b8606bd040d59409203acc178c14fe4ec4d6b38e8955d87dc4ecea1R666 for that purpose.

If this is not the way to go to know how many items match the current query... then I could really use your optimizing skills to do it right :pleading_face: :pray: :innocent:

Reply:
> I really think we need to pull out the filter parsing and fail the request if these are included

Oh! didn't get your update while I was typing. Sure! I can try to make the request fail when those filters are present (just in the particular case of the count being requested).

Reply:
> If this is not the way to go to know how many items match the current query... then I could really use your optimizing skills to do it right

puh, I don't know if it can be optimized further. I would rather think we should aggressively limit how often we have to run the query. Let's just go with it for now, we should know soon enough if this will be a problem.

Reply:
Done in [c6bbbb1](https://github.com/galaxyproject/galaxy/pull/13555/commits/c6bbbb13ee6aba47cee7251926d528de50132380) :)